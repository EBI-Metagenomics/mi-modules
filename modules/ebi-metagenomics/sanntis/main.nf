process SANNTIS {
    tag "$meta.id"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'biocontainers/sanntis:0.9.3.5--pyhdfd78af_1' :
        'biocontainers/sanntis:0.9.3.5--pyhdfd78af_1' }"

    input:
    tuple val(meta), path(interproscan_output), path(gbk)

    output:
    tuple val(meta), path("*_sanntis.gff"), emit: gff
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    grep -v "/protein_id=" ${gbk} > ${prefix}_prepped.gbk
    sanntis \
    --ip-file ${interproscan_output} \
    --outfile ${prefix}_sanntis.gff \
    --cpu ${task.cpus} \
    ${args} \
    ${prefix}_prepped.gbk


    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        SanntiS: \$(sanntis --version | sed "s/SanntiS\\ //g")
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_sanntis.gff

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        SanntiS: \$(sanntis --version | sed "s/SanntiS\\ //g")
    END_VERSIONS
    """
}
