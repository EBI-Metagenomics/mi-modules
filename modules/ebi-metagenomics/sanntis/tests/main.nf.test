nextflow_process {

    name "Test Process SANNTIS"
    script "../main.nf"
    process "SANNTIS"

    tag "modules"
    tag "modules_nfcore"
    tag "sanntis"

    test("test_sanntis") {

        when {
            process {
                """
                input[0] = [
                    [ id:'MGYG000300113' ], // meta map
                    file("${moduleDir}/tests/fixtures/IPS.subset.tsv", checkIfExists: true),
                    file("${moduleDir}/tests/fixtures/MGYG000300113.gbk", checkIfExists: true)
                    ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_sanntis_stub") {
        options '-stub'

        when {
            process {
                """
                input[0] = [
				    [ id:'MGYG000300113' ], // meta map
				    file("${moduleDir}/tests/fixtures/IPS.subset.tsv"),
				    file("${moduleDir}/tests/fixtures/MGYG000300113.gbk")
				]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}
