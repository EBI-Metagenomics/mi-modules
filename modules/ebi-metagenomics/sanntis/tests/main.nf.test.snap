{
    "test_sanntis": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "MGYG000300113"
                        },
                        "MGYG000300113_sanntis.gff:md5,587842de5d5ca8b731b97f03bec58101"
                    ]
                ],
                "1": [
                    "versions.yml:md5,14d0b69d48fa9993cb4602246082d0de"
                ],
                "gff": [
                    [
                        {
                            "id": "MGYG000300113"
                        },
                        "MGYG000300113_sanntis.gff:md5,587842de5d5ca8b731b97f03bec58101"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,14d0b69d48fa9993cb4602246082d0de"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.04.5"
        },
        "timestamp": "2024-09-05T09:50:25.397800952"
    },
    "test_sanntis_stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "MGYG000300113"
                        },
                        "MGYG000300113_sanntis.gff:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,14d0b69d48fa9993cb4602246082d0de"
                ],
                "gff": [
                    [
                        {
                            "id": "MGYG000300113"
                        },
                        "MGYG000300113_sanntis.gff:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,14d0b69d48fa9993cb4602246082d0de"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.04.5"
        },
        "timestamp": "2024-09-05T09:50:32.439151633"
    }
}